### PR TITLE
KR test endpoints

### PR DESCRIPTION
**Pull request by kjreczek**
I added tests to section - some endpoints, new model for test (in section file), service and controller separate for tests. 
Routes and Repository are the same for section and tests.

post - **/sections/:id/test** - add test
patch - **/sections/:sectionId/test/:testId** - update test
delete - **/sections/:sectionId/test/:testId** - delete test

### CODE REVIEW

File: Api/Src/Controllers/TestController.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        await this.testService.updateTest(sectionId, testId, test);
+        res.status(200).json({message: 'Test was updated'});
+    }
+
+    deleteTest = async (req: Request, res: Response) => {
+        const sectionId = new mongoose.Types.ObjectId(req.params.sectionId);
+        const testId = new mongoose.Types.ObjectId(req.params.testId);
+        const section = await this.sectionService.getSectionById(sectionId);
+        if (!section) res.status(404).json({message: 'Section not found'});
+        await this.testService.deleteTest(sectionId, testId);
```
Comment (fszwedo) [inline Api/Src/Controllers/TestController.ts:38]: some condition to check if test exists would be good here - at the moment removing non-existing test ends with 200

File: Api/Src/Controllers/TestController.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        res.status(200).json({message: 'Test added to section'});
+    }
+
+    updateTest = async (req: Request, res: Response) => {
+        const sectionId = new mongoose.Types.ObjectId(req.params.sectionId);
+        const testId = new mongoose.Types.ObjectId(req.params.testId);
+        const test = req.body;
+        const section = await this.sectionService.getSectionById(sectionId);
+        if (!section) res.status(404).json({message: 'Section not found'});
+        await this.testService.updateTest(sectionId, testId, test);
```
Comment (fszwedo) [inline Api/Src/Controllers/TestController.ts:29]: some condition to check if the test exists might be good here, at the moment updating non-existing test ends with 200

**Review by JustynaSmoliga — COMMENTED** [review]
I have tested endpoints and after updating the test, I get the message: "Test has been updated", but when I GET the section containing this test, there is an old test (instead of a new test).

**Review by fszwedo — COMMENTED** [review]
> I have tested endpoints and after updating the test, I get the message: "Test has been updated", but when I GET the section containing this test, there is an old test (instead of a new test).

same here :(

File: Api/Src/Controllers/TestController.ts (hunk 1 above)
Comment (kjreczek) [reply Api/Src/Controllers/TestController.ts]: Right, I added new condition to check if test exists.

File: Api/Src/Controllers/TestController.ts (hunk 2 above)
Comment (kjreczek) [reply Api/Src/Controllers/TestController.ts]: Right, I added new condition to check if test exists.

**Comment by kjreczek** [comment]
@JustynaSmoliga @fszwedo i fixed update and now it should work.

**Comment by JustynaSmoliga** [comment]
After fixes everything works ;)

**Review by JustynaSmoliga — APPROVED** [review]

**Comment by fszwedo** [comment]
> @JustynaSmoliga @fszwedo i fixed update and now it should work.

@kjreczek for me unfortunately the update endpoint throws 404 all the time with test not found :(
example: http://localhost:5000/api/sections/604e3a6213bc103ee0608f64/test/604e3d69db589b4cb8fd2b0e

could you take a look if it works ok for you?

**Comment by kjreczek** [comment]
@fszwedo i hope now it is working

**Review by fszwedo — APPROVED** [review]
all good now, thanks :)

**Pull request merged**